pkg/cloudprovider/providers/gce/gce_clusters.go

```diff
@@ -42,13 +43,19 @@ func (gce *GCECloud) ListClusters(ctx context.Context) ([]string, error) {
 
 func (gce *GCECloud) GetManagedClusters(ctx context.Context) ([]*container.Cluster, error) {
 	managedClusters := []*container.Cluster{}
-	for _, zone := range gce.managedZones {
-		clusters, err := gce.getClustersInZone(zone)
-		if err != nil {
-			return nil, err
-		}
-		managedClusters = append(managedClusters, clusters...)
+	var location string
+	if len(gce.managedZones) > 1 {
+		// Mutliple managed zones means this is a regional cluster
+		// so use the regional location and not the zone.
+		location = gce.region
+	} else if len(gce.managedZones) == 1 {
+		location = gce.managedZones[0]
 	}
+	clusters, err := gce.getClustersInLocation(location)
+	if err != nil {
+		return nil, err
+	}
+	managedClusters = append(managedClusters, clusters...)
```

Comment:
do we need to make managedClusters? Why not just return clusters directly?

Reply:
Sorry leftover from the old code. Cleaned up.